test/services/geolocation_service_test.dart

```diff
@@ -0,0 +1,166 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:geolocator/geolocator.dart";
+import "package:mockito/mockito.dart";
+import "package:proxima/services/geolocation_service.dart";
+import "package:test/test.dart";
+
+/// Helper function to create a simple position object
+Position getSimplePosition(double latitude, double longitude) {
+  return Position(
+    latitude: latitude,
+    longitude: longitude,
+    timestamp: DateTime.fromMillisecondsSinceEpoch(123123123),
+    accuracy: 0,
+    altitude: 0,
+    heading: 0,
+    speed: 0,
+    speedAccuracy: 0,
+    altitudeAccuracy: 0,
+    headingAccuracy: 0,
+  );
+}
+
+/// We need to override the methods in order to mock them in the tests
+class MockGeolocatorPlatform extends Mock implements GeolocatorPlatform {
+  @override
+  Future<bool> isLocationServiceEnabled() async {
+    return super.noSuchMethod(
+      Invocation.method(#isLocationServiceEnabled, []),
+      returnValue: Future.value(true),
+    );
+  }
+
+  @override
+  Future<LocationPermission> checkPermission() async {
+    return super.noSuchMethod(
+      Invocation.method(#checkPermission, []),
+      returnValue: Future.value(LocationPermission.always),
+    );
+  }
+
+  @override
+  Future<LocationPermission> requestPermission() async {
+    return super.noSuchMethod(
+      Invocation.method(#requestPermission, []),
+      returnValue: Future.value(LocationPermission.always),
+    );
+  }
+
+  @override
+  Future<Position> getCurrentPosition({
+    LocationSettings? locationSettings,
+  }) async {
+    return super.noSuchMethod(
+      Invocation.method(#getCurrentPosition, []),
+      returnValue: Future.value(getSimplePosition(0, 0)),
+    );
+  }
+}
+
+void main() {
+  group("GeoLocationService", () {
+    late GeoLocationService geoLocationService;
+    late MockGeolocatorPlatform mockGeolocator;
+
+    setUp(() {
+      mockGeolocator = MockGeolocatorPlatform();
+      geoLocationService = GeoLocationService(geoLocator: mockGeolocator);
+    });
+
+    test("getCurrentPosition returns GeoPoint", () async {
+      const latitude = 37.4219983;
+      const longitude = -122.084;
+      const expectedGeoPoint = GeoPoint(latitude, longitude);
+
+      when(mockGeolocator.isLocationServiceEnabled())
+          .thenAnswer((_) async => true);
+      when(mockGeolocator.checkPermission())
+          .thenAnswer((_) async => LocationPermission.always);
```

Comment:
Isn't this a repetition with lines 26-39?
I would personally think that the `when` should be prefered over the `noSuchMethod` since it is more readable (if it is possible to do once and for all in the setup). Note that, changing `MockGeolocatorPlatform` would also mean changing `post_repository_test.dart::MockGeoLocationService`.

Reply:
It is indeed necessary, the overrides with the `noSuchMethod` are a boiler plate needed in order mock the result with `when`. 
If those are removed, the function won't be mocked properly and a bad state exception will be thrown by the mocking service.

Also, I am not entirely sure of my understanding of `noSuchMethod` but I think that if they are called without having mocking it with `when` priorly, they will throw an exception. This what I have noticed while coding the tests.

So I think that we need to let those here in order for the mocking to work properly.

Reply:
Oh, we learn every days! Thank you, and sorry for my mistake :)